### DISCUSSION TITLE
issue: useFieldArrays rules validate messes up ordering of errors

### DISCUSSION THREAD

**Discussion by nithiwatter** [react-hook-form/react-hook-form#8976] [Q&A]
### Version Number

7.34.2

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-usefieldarray-forked-iyejbp?file=/src/index.js

### Steps to reproduce

1. Go to the example sandbox provided by the docs.
2. Add a rules validate (ex. minLength) to `useFieldArrays`
3. Try appending two times
4. Observe that the errors are messed up in terms of ordering

<img width="902" alt="image" src="https://user-images.githubusercontent.com/59894785/188256546-b40610ef-db3f-47c8-8c79-c794a0a08daa.png">
<img width="792" alt="image" src="https://user-images.githubusercontent.com/59894785/188256557-fa0ae5da-86f2-443d-858a-7c1a8dd371aa.png">

### Expected behaviour

The latest error log in console should be of correct order (ex. undefined for the first input and present for the last two inputs).

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
This is consider as a limtiatino as when the `root` object injects into an array object, it turns the errors into an object instead.

```tsx
//before

const errors = [{}, {}]
```

```tsx
//after

const errors = {0: {}, 1: {}, root: {}}
```

**Reply by nithiwatter** [reply]
So is there a way around this? I am currently displaying the errors and the ordering of the ui gets messed up + confusing. Should I just avoid using the new rules/validate? What would be your suggestion for an alternative? Thank you!

**Reply by bluebill1049** [reply]
perhaps you want to write a function to iterator through it and produce an array instead?
